internal/gitserver/protocol/gitserver.go

```diff
@@ -260,6 +260,25 @@ type ExecRequest struct {
 	NoTimeout      bool     `json:"noTimeout"`
 }
 
+type DiskInfoResponse struct {
+	FreeSpace  uint64 `json:"free_space"`
+	TotalSpace uint64 `json:"total_space"`
+}
+
+func (di *DiskInfoResponse) ToProto() *proto.DiskInfoResponse {
+	return &proto.DiskInfoResponse{
+		FreeSpace:  di.FreeSpace,
+		TotalSpace: di.TotalSpace,
+	}
+}
+
+func (di *DiskInfoResponse) FromProto(p *proto.DiskInfoResponse) {
+	*di = DiskInfoResponse{
+		FreeSpace:  p.FreeSpace,
+		TotalSpace: p.TotalSpace,
```

Comment:
minor: generally, accessing proto fields should be done with the generated getters/setters because messages are all pointers, which may be `nil`.

```suggestion
		FreeSpace:  p.GetFreeSpace(),
		TotalSpace: p.GetTotalSpace(),
```